rpcserver.go

```diff
@@ -2441,7 +2442,14 @@ func (r *rpcServer) sendPayment(stream *paymentStream) error {
 		case <-r.quit:
 			return errors.New("rpc server shutting down")
 
-		case payIntent := <-payChan:
+		case payIntent, ok := <-payChan:
+			// If the receive loop is done, we break the send loop
+			// and wait for the ongoing payments to finish before
+			// exiting.
+			if !ok {
+				break sendLoop
```

Comment:
is returning equivalent?

Reply:
With the following commit it will no longer be, as we wait for goroutines to exit.